lib/cc/analyzer/engine.rb

```diff
@@ -74,18 +74,24 @@ def qualified_name
       end
 
       def container_options
-        [
+        options = [
           "--cap-drop", "all",
           "--label", "com.codeclimate.label=#{@label}",
           "--log-driver", "none",
-          "--memory", metadata["memory"].to_s,
           "--memory-swap", "-1",
           "--net", "none",
           "--rm",
           "--volume", "#{code.host_path}:/code:ro",
           "--volume", "#{config_file.host_path}:/config.json:ro",
           "--user", "9000:9000"
         ]
+        if metadata["memory"]
+          memory = metadata["memory"].to_s
+          unless memory.empty?
+            options.concat ["--memory", memory]
+          end
+        end
```

Comment:
You might be able to simplify this a bit using activesupport. Not sure you'd need to `to_s` with `#present?`.

```ruby
if (memory = metadata["memory"]).present?
  options.concat(["--memory", memory])
end
```

Reply:
Aah, I wasn't sure if it was safe to rely on ActiveSupport in that part of the code.

Reply:
Yep, I'm pretty sure it's available.

Reply:
Changed it, though a `to_s` is still necessary, because the specs check that it's a String.